src/NuGet.Clients/NuGet.PackageManagement.UI/NuGetPackageManagerControlSearchTask.cs

```diff
@@ -26,21 +27,39 @@ public NuGetPackageManagerControlSearchTask(PackageManagerControl packageManager
         }
         public void Start()
         {
-            SetStatus(VsSearchTaskStatus.Started);
-            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            IDisposable activity = _packageManagerControl._pmuiGestureintervalTracker.Start(nameof(NuGetPackageManagerControlSearchTask));
+            var activityIsDisposed = false;
+
+            try
             {
-                await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                SetStatus(VsSearchTaskStatus.Started);
+                NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                // Set a new cancellation token source which will be used to cancel this task in case
-                // new loading task starts or manager ui is closed while loading packages.
-                var loadCts = new CancellationTokenSource();
-                var oldCts = Interlocked.Exchange(ref _packageManagerControl._loadCts, loadCts);
-                oldCts?.Cancel();
-                oldCts?.Dispose();
+                    activityIsDisposed = true;
+                    using (activity)
+                    {
+                        // Set a new cancellation token source which will be used to cancel this task in case
+                        // new loading task starts or manager ui is closed while loading packages.
+                        var loadCts = new CancellationTokenSource();
+                        var oldCts = Interlocked.Exchange(ref _packageManagerControl._loadCts, loadCts);
+                        oldCts?.Cancel();
+                        oldCts?.Dispose();
 
-                await _packageManagerControl.SearchPackagesAndRefreshUpdateCountAsync(searchText: _searchQuery.SearchString, useCachedPackageMetadata: true, pSearchCallback: _searchCallback, searchTask: this);
-                SetStatus(VsSearchTaskStatus.Completed);
-            }).PostOnFailure(nameof(NuGetPackageManagerControlSearchTask));
+                        await _packageManagerControl.SearchPackagesAndRefreshUpdateCountAsync(searchText: _searchQuery.SearchString, useCachedPackageMetadata: true, pSearchCallback: _searchCallback, searchTask: this);
+                        SetStatus(VsSearchTaskStatus.Completed);
+                    }
+                }).PostOnFailure(nameof(NuGetPackageManagerControlSearchTask));
+            }
+            finally
```

Comment:
What if there's an exception before the JTF.RunAsync calls? I attempted to add a finally block to ensure the activity gets disposed. I believe this could dispose the activity before the async method uses the activity, so I may need to refactor with a better pattern.

Reply:
Would it work to just put the tracker inside the `RunAsync()` with a `using`?  We only care about how long the stuff inside the `RunAsync()` took right?

Reply:
> Probably before the JTF.RunAsync has even started.

I assumed so, but wasn't sure of which pattern to go with (and wanted to start getting feedback on the meat of the PR).

> We only care about how long the stuff inside the RunAsync() took right?

Yes, but I'm trying to guarantee that the timer doesn't run forever if it never makes it to the `using`. Maybe I get that guarantee for free now that I think about it: If anything throws before the JTF, or when transitioning to the RunAsync, I believe the object gets disposed anyway.

Reply:
Removed manual Dispose logic.